### PR TITLE
update ng-packagr to version 4

### PR DESCRIPTION
**Pull request by alan-agius4**
update `ng-packagr` to version 4

ng-packagr v4 requires that enableResourceInlining needs to be enabled for libraries that contain components

Relates also to https://github.com/angular/angular-cli/pull/11358

### CODE REVIEW

File: tests/angular_devkit/build_ng_packagr/ng-packaged/projects/lib/tsconfig.lib.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
     "fullTemplateTypeCheck": true,
     "strictInjectionParameters": true,
     "flatModuleId": "AUTOGENERATED",
-    "flatModuleOutFile": "AUTOGENERATED"
+    "flatModuleOutFile": "AUTOGENERATED",
+    "enableResourceInlining": true
```
Comment (alan-agius4) [inline tests/angular_devkit/build_ng_packagr/ng-packaged/projects/lib/tsconfig.lib.json]: We need to properly communicate that if consumers are using `ng-packagr v3` when they update to `v4` they need to add this flag in their existing tsconfig.

Comment (filipesilva) [reply tests/angular_devkit/build_ng_packagr/ng-packaged/projects/lib/tsconfig.lib.json]: WDYT of checking that this field is missing and adding a warning/error at runtime?

Comment (alan-agius4) [reply tests/angular_devkit/build_ng_packagr/ng-packaged/projects/lib/tsconfig.lib.json]: I was actually thinking of switching this option to true by default in ng-packagr

Comment (filipesilva) [reply tests/angular_devkit/build_ng_packagr/ng-packaged/projects/lib/tsconfig.lib.json]: That works too, and seems like you have already 👍 https://github.com/dherges/ng-packagr/pull/1021

Comment (alan-agius4) [reply tests/angular_devkit/build_ng_packagr/ng-packaged/projects/lib/tsconfig.lib.json]: yeah :)

**Pull request merged**